dsl/attr_reader.cc

```diff
@@ -21,7 +23,20 @@ pair<core::NameRef, core::Loc> getName(core::MutableContext ctx, ast::Expression
             ENFORCE(loc.source(ctx).size() > 1 && loc.source(ctx)[0] == ':');
             loc = core::Loc(loc.file(), loc.beginPos() + 1, loc.endPos());
         } else if (lit->isString(ctx)) {
-            res = lit->asString(ctx);
+            core::NameRef nameRef = lit->asString(ctx);
+            auto shortName = nameRef.data(ctx)->shortName(ctx);
+            bool validAttr =
+                regex_match(shortName.begin(), shortName.end(), basic_regex("([:alpha:]|_)([:alnum:]|_)*"));
```

Comment:
c++ regexes are notoriously bad. Please use an explicit loop :-)

Reply:
See https://github.com/sorbet/sorbet/commit/81b567ddd90c087d77d948c10aef89d40f5343f8 for prior example